tests/providers/amazon/aws/operators/test_s3_bucket.py

```diff
@@ -0,0 +1,70 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import os
+import unittest
+
+from moto import mock_s3
+
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.providers.amazon.aws.operators.s3_bucket import S3CreateBucketOperator, S3DeleteBucketOperator
+
+BUCKET_NAME = os.environ.get("BUCKET_NAME", "test-airflow-bucket")
+TASK_ID = os.environ.get("TASK_ID", "test-s3-operator")
+
+
+class TestS3CreateBucketOperator(unittest.TestCase):
+    @mock_s3
+    def test_execute_if_bucket_exist(self):
+        mock_hook = S3Hook()
+        # execute s3 bucket create operator
+        op = S3CreateBucketOperator(bucket_name=BUCKET_NAME, task_id=TASK_ID)
+        op.execute({})
+        # assert if the bucket has been created
+        self.assertTrue(mock_hook.check_for_bucket(bucket_name=BUCKET_NAME))
```

Comment:
Please assert that the mock has been called `mock_hook.check_for_bucket.assert_called_once_with` - not call it by yourself. It should be called by/inside the `S3CreateBucketOperator.execute`.

Same goes for your other tests.

Reply:
The important call is the `mock_hook.create_bucket` !

Reply:
For more information see the docs here: https://docs.python.org/3.7/library/unittest.mock.html

If you need some help let me know :)

Reply:
Oh right got it you want me to assert calls!! I thought asserting the final state would suffice!